sklearn/linear_model/_coordinate_descent.py

```diff
@@ -1075,6 +1075,16 @@ def fit(self, X, y, sample_weight=None, check_input=True):
         # workaround since _set_intercept will cast self.coef_ into X.dtype
         self.coef_ = np.asarray(self.coef_, dtype=X.dtype)
 
+        # check for finiteness of coefficients
+        num_nonfinite = (~np.isfinite(self.coef_)).sum() + (
+            ~np.isfinite(self.intercept_)
+        ).sum()
+        if num_nonfinite > 0:
+            raise ValueError(
+                f"Coordinate descent iterations resulted in {num_nonfinite} non-finite"
+                " parameter values."
+            )
```

Comment:
I do not think the count is actionable. I think we can have an error message similar to my suggestion in: https://github.com/scikit-learn/scikit-learn/pull/22149/files#r780722680